src/components/transport_manager/src/usb/libusb/usb_connection.cc

```diff
@@ -129,11 +129,6 @@ std::string hex_data(const unsigned char* const buffer,
 
 void UsbConnection::OnInTransfer(libusb_transfer* transfer) {
   SDL_LOG_AUTO_TRACE();
-  if (transfer->status == LIBUSB_TRANSFER_CANCELLED) {
-    SDL_LOG_DEBUG("Free already canceled transfer.");
-    libusb_free_transfer(transfer);
-    return;
-  }
```

Comment:
@RomanReznichenkoLuxoft I would recommend refactoring of this function instead:
```
void UsbConnection::OnInTransfer(libusb_transfer* transfer) {
  SDL_LOG_AUTO_TRACE();
  SDL_LOG_TRACE("enter with Libusb_transfer*: " << transfer);
  switch (transfer->status) {
    case LIBUSB_TRANSFER_COMPLETED: {
      SDL_LOG_DEBUG("USB incoming transfer, size:"
                    << transfer->actual_length << ", data:"
                    << hex_data(transfer->buffer, transfer->actual_length));
      ::protocol_handler::RawMessagePtr data(new protocol_handler::RawMessage(
          0, 0, in_buffer_, transfer->actual_length, false));
      controller_->DataReceiveDone(device_uid_, app_handle_, data);
      break;
    }

    case LIBUSB_TRANSFER_CANCELLED: {
      SDL_LOG_DEBUG("Free already canceled transfer.");
      break;
    }

    default: {
      SDL_LOG_ERROR("USB incoming transfer failed: "
                    << libusb_error_name(transfer->status));
      controller_->DataReceiveFailed(
          device_uid_, app_handle_, DataReceiveError());
    }
  }

  if (waiting_in_transfer_cancel_) {
    libusb_free_transfer(transfer);
    waiting_in_transfer_cancel_ = false;
    return;
  }

  if (!PostInTransfer()) {
    SDL_LOG_ERROR("USB incoming transfer failed with "
                  << "LIBUSB_TRANSFER_NO_DEVICE. Abort connection.");
    AbortConnection();
  }
}
```

And for out transfer:
```
void UsbConnection::OnOutTransfer(libusb_transfer* transfer) {
  SDL_LOG_AUTO_TRACE();
  SDL_LOG_TRACE("enter with  Libusb_transfer*: " << transfer);
  auto error_code = TransportAdapter::OK;
  {
    sync_primitives::AutoLock locker(out_messages_mutex_);
    switch (transfer->status) {
      case LIBUSB_TRANSFER_COMPLETED: {
        bytes_sent_ += transfer->actual_length;
        if (current_out_message_->data_size() == bytes_sent_) {
          SDL_LOG_DEBUG(
              "USB out transfer, data sent: " << current_out_message_.get());
          controller_->DataSendDone(
              device_uid_, app_handle_, current_out_message_);
          error_code = PopOutMessage();
        }
        break;
      }

      case LIBUSB_TRANSFER_CANCELLED: {
        SDL_LOG_DEBUG("Free already canceled transfer.");
        break;
      }

      default: {
        SDL_LOG_ERROR(
            "USB out transfer failed: " << libusb_error_name(transfer->status));
        controller_->DataSendFailed(
            device_uid_, app_handle_, current_out_message_, DataSendError());
        error_code = PopOutMessage();
      }
    }

    if (waiting_out_transfer_cancel_ || current_out_message_.use_count() == 0) {
      libusb_free_transfer(transfer);
      out_transfer_ = nullptr;
      waiting_out_transfer_cancel_ = false;
      return;
    }
  }

  if (TransportAdapter::FAIL == error_code) {
    AbortConnection();
  }
}
```

Reply:
done